src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -98,7 +101,24 @@ public int getHttpStatus() {
   }
 
   public boolean isSuccess() {
-    return successCodes.contains(getHttpStatus());
+    boolean isSuccess = false;
+    if (checkRetriable()) {
+      isSuccess = false;
+    } else if (successCodes.contains(getHttpStatus())) {
+      isSuccess = true;
+    }
+    return isSuccess;
+  }
+
+  private boolean checkRetriable() {
+    Gson gson = new Gson();
+    JsonObject jo = gson.fromJson(this.getResponseBody(), JsonObject.class);
+    if (jo.get("status") != null && jo.get("status").getAsString().equals("failure")) {
```

Comment:
define keywords like status failure or error message as String literals and use them.
Merge these two for loops into one.

Reply:
Defined String Literals and merged two **if** blocks a752f6b05f68ae64ce9ed63c94fc7b1cc1865b12